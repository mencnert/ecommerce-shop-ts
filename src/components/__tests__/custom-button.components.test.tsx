import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import CustomButton from "../custom-button/custom-button.component";

describe("Test button", () => {
  it("should click", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <CustomButton type="button" isGoogleSignIn={true} onClick={onClick} />
    );
    wrapper.find("button").simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});

describe("snapshots", () => {
  it('is type "button" and is goole sign in', () => {
    const wrapper = shallow(
      <CustomButton type="button" isGoogleSignIn={true} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('is type "submit" and is not goole sign in', () => {
    const wrapper = shallow(
      <CustomButton type="submit" isGoogleSignIn={false} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
