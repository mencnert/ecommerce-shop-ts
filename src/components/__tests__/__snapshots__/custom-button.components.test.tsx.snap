// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots is type "button" and is goole sign in 1`] = `
<button
  className="google-sign-in custom-button"
  type="button"
/>
`;

exports[`snapshots is type "submit" and is not goole sign in 1`] = `
<button
  className=" custom-button"
  type="submit"
/>
`;
